diagnostic_rt_qpcr/operation_types/aliquot_primer_probe/protocol.rb

```diff
@@ -1,32 +1,137 @@
-# frozen_string_literal: true
+# Aliquot Primer Probe
+# Written By Dany Fu 2020-05-05
 
-# This is a default, one-size-fits all protocol that shows how you can
-# access the inputs and outputs of the operations associated with a job.
-# Add specific instructions for this protocol!
+needs 'Standard Libs/LabwareNames'
 
+# 1) Upon receipt, store dried primers and probes at 2-8C.
+# 2) Precautions: These reagents should only be handled in a clean area and
+# stored at appropriate temperatures (see below) in the dark. Freeze-thaw cycles
+# should be avoided. Maintain cold when thawed.
+# 3) Using aseptic technique, suspend dried reagents in 1.5 mL of nuclease-free
+# water (50X working concentration) and allow to rehydrate for 15 min at room
+# temperature in the dark.
+# 4) Mix gently and aliquot primers/probe in 300 µL volumes into 5 pre-labeled
+# tubes. Store a single aliquot of primers/probe at 2-8oC in the dark. Do not
+# refreeze (stable for up to 4 months). Store remaining aliquots at <= -20oC
+# in a non-frost-free freezer.
 class Protocol
+  include Units
+  include LabwareNames
+
+  OUTPUT_ITEMS_NUM = { qty: 5, units: TUBE_MICROFUGE }.freeze
+  TIME_REHYDRATE = { qty: 15, units: MINUTES }.freeze
+  VOL_WATER = { qty: 1.5, units: MILLILITERS }.freeze
+  VOL_SUSPENSION = { qty: 300, units: MICROLITERS }.freeze
+
+  def default_job_params
+    {
+      cold_room: 'M4' # location wizard of cold room at DAMPLab
+    }
+  end
 
   def main
+    get_tubes(count: operations.length)
+    operations.retrieve
 
-    operations.retrieve.make
+    save_output(operations)
+    suspend_primer_mix
 
-    tin  = operations.io_table 'input'
-    tout = operations.io_table 'output'
+    # Group the operations by the input reagent
+    ops_by_input = operations.group_by { |op| op.input('Primer Set').item }
+    ops_by_input.each do |primer, ops|
+      make_aliquots(ops: ops, primer: primer)
+    end
 
+    operations.store(interactive: true, io: 'output', method: 'boxes')
+  end
+
+  # Get 5 1.5 mL tubes per dried reagent
+  # @param count [Integer] the number of operations currently running
+  def get_tubes(count:)
     show do
-      title 'Input Table'
-      table tin.all.render
+      title "Get new #{TUBE_MICROFUGE}"
+      check "Please get #{count * OUTPUT_ITEMS_NUM[:qty]} #{TUBE_MICROFUGE}"
+    end
+  end
+
+  # Create and save multiple output Items per operation
+  # @param operations [OperationList] List of operations
+  def save_output(operations)
```

Comment:
This and the parallel method in the templates, as well as the associated methods, should be extracted into a module.